autometa/taxonomy/ncbi.py

```diff
@@ -378,10 +385,15 @@ def is_common_ancestor(self, taxid_A, taxid_B):
         common_ancestor.discard(1)  # This discards root
         return True if common_ancestor else False
 
-    def is_valid_taxid(self, taxid):
+    def convert_taxid_dtype(self, taxid):
         """
-        Checks if the given `taxid` is a positive integer or not as well as if it is present in nodes.dmp or names.dmp
-        In case the taxid provided is in the form of a string the function will cast the string as an integer
+        Checks if the given `taxid` is a positive integer.
+        In case the `taxid` provided is in the form of a string the function will cast the string as an integer.
+        Checks if the `taxid` is present in either nodes.dmp or names.dmp.
```

Comment:
Could you make this more readable? Also, the function converts the taxid with whatever type provided to an integer, however the docstring is talking about checks. These checks should probably be discussed _after_ the first sentence. Keep the first concise as to the use of the function then describe the details later.